eve/main.yml

```diff
@@ -5,6 +5,256 @@ branches:
+  - ShellCommand: &build_all
+      name: Build everything
+      env:
+        PYTHON_SYS: python3.6
+      # There are 3 CPUs available for Docker, and 1 for `doit` in the Pod.
+      # Given the network IO-bound nature of some of the build steps, and
+      # most build steps running in Docker, set concurrency to 4.
+      command: ./doit.sh -n 4
+      usePTY: true
+      haltOnFailure: true
+  - ShellCommand: &ssh_ip_setup
+      name: Install SSH keys and report connection info
+      command: |
+        mkdir -p ~/.ssh
+        echo "%(secret:ssh_pub_keys)s" >> ~/.ssh/authorized_keys
+        IP=$(
+          ip -f inet addr show eth0 | sed -En 's/^.*inet ([0-9.]+).*$/\1/p'
+        )
+        cat << END
+        Connect to this worker using:
+            ssh eve@$IP
+        END
+  - ShellCommand: &check_iso_checksum
+      name: Check image with checksum
+      command: sha256sum -c SHA256SUM
+      haltOnFailure: true
+  - Upload: &upload_artifacts
+      name: Upload artifacts
+      source: artifacts
+  - Upload: &upload_report_artifacts
+      name: Upload sosreport logs to artifacts
+      source: sosreport
+  - ShellCommand: &collect_sosreport
+      name: Collect logs using sosreport
+      command: >
+        sudo sosreport --all-logs
+        -o metalk8s -kmetalk8s.podlogs=True
+        -o containerd -kcontainerd.all=True -kcontainerd.logs=True
+        --batch --tmp-dir /var/tmp &&
+        sudo chown eve:eve /var/tmp/sosreport*
+  - ShellCommand: &wait_debug
+      name: Debug step - wait before allowing resource destruction
+      env:
+        # Use this step with STEP_NAME and DURATION environment values to
+        # customize the command behaviour. DURATION default value is 3600
+        # (wait duration, in seconds).
+        STEP_NAME: default
+      timeout: 3600
+      command: |
+        bash -c '
+        DEBUG_STEPS="%(prop:debug)s"
+        DURATION="${DURATION:-3600}"
+        RUN_STEP=0
+        REASON=""
+        if [ -z "$DEBUG_STEPS" ]; then
+          REASON="\"debug\" build property not set"
+        elif [ "$DEBUG_STEPS" = all ]; then
+          RUN_STEP=1
+          REASON="\"debug\" property set to \"all\""
+        elif [[ "$DEBUG_STEPS" =~ ^[a-z\-]+(~[a-z\-]+)*$ ]]; then
+          IFS="~" read -ra SELECTED <<< "$DEBUG_STEPS"
+          for selected in "${SELECTED[@]}"; do
+            if [ "$selected" = "$STEP_NAME" ]; then
+              RUN_STEP=1
+              REASON="step selected in \"$DEBUG_STEPS\""
+              break
+            fi
+          done
+          if [ "$RUN_STEP" -eq 0 ]; then
+            REASON="step not in \"$DEBUG_STEPS\""
+          fi
+        else
+          REASON="invalid \"debug\" property value"
+          cat >&2 << EOF
+        Invalid "debug" build property value "$DEBUG_STEPS".
+        Must use either:
+          - "all", to select all debug steps
+          - a single step name
+          - a list of step names, separated by tilde signs "~",
```

Comment:
Weird choice `~` as separation character between step names.
I would have used commas or semicolons, but ok, it's a detail. :)
Otherwise, I like this debugging system, way more usable than the previous one.